backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -311,13 +306,16 @@ private List<Integer> getOnlineCores(List<VdsCpuUnit> cpuTopology) {
     private List<VdsCpuUnit> getMaxFreedSocket(List<VdsCpuUnit> cpuTopology) {
         List<VdsCpuUnit> chosenSocket = Collections.emptyList();
         List<VdsCpuUnit> temp;
+        List<Integer> onlineSockets = getOnlineSockets(cpuTopology);
+        int socketId = onlineSockets.get(0);
         for (int socket : getOnlineSockets(cpuTopology)) {
             temp = getFreeCpusInSocket(cpuTopology, socket);
             if (temp.size() > chosenSocket.size()) {
                 chosenSocket = temp;
+                socketId = socket;
             }
         }
-        return chosenSocket;
+        return getCoresInSocket(cpuTopology, socketId);
```

Comment:
Do you think we could use streams to find the socket with max number of cores? I've found this on SO and it seems it could work
```
lists.stream().max(Comparator.comparing(List::size)).get()
```

Reply:
but that isn't enough, is it? we need to filter only available CPUs before comparing between such lists. at the end to return the list without that filter.

Reply:
getCoresInSocket returns only free cores?

Reply:
> getCoresInSocket returns only free cores?

no, it returns all of them in the given socket ID, `getFreeCpusInSocket` however, does filter out taken CPUs..

Reply:
the real difference is that once we called `getMaxFreedSocket`, previously it returned the socket using `getFreeCpusInSocket`, which returns the socket after removing taken CPUs. In the isolate-threads we try to filter CPUs based on:
```
if (cpusInCore.stream().anyMatch(VdsCpuUnit::isPinned)) {
    continue;
}
```
of course non of them are pinned based on the above, so the core is good to be used, which is wrong.

Now we will get the whole socket including every CPU within, this will work as expected.